### PR TITLE
SQL highlights update

### PR DESCRIPTION
**Pull request by matthias-Q**
Add recently added keywords
restructures usage of nodes a bit. Most of them where just used as `@keyword`
I have grouped some keywords as `@attributes` and `@conditionals`

### CODE REVIEW

**Comment by matthias-Q** [comment]
@theHamsta do you need a before after screenshot?

File: queries/sql/highlights.scm [hunk 1]
```diff
@@ -27,13 +27,61 @@
...
+    parameter: [(literal)]?)))
+
 (literal) @string
 (comment) @comment
 (marginalia) @comment
 
 ((literal) @number
  (#lua-match? @number "^%d+$"))
 
+(parameter) @variable
```
Comment (theHamsta) [inline queries/sql/highlights.scm:42]: Can you tell us why `@parameter` doesn't apply here?

**Review by theHamsta — APPROVED** [review]

**Comment by theHamsta** [comment]
@derekstride

Thank you! Screenshots are typically not needed. Queries should just be logical and consistent with other languages

File: queries/sql/highlights.scm (hunk 1 above)
Comment (matthias-Q) [reply queries/sql/highlights.scm:42]: Yes, you are right. I will change this to a `@parameter` this evening.

**Review by DerekStride — APPROVED** [review]
Thanks @matthias-Q!

**Comment by matthias-Q** [comment]
I have changed `@variable` to `@parameter` and added `@float` which was merged upstream yesterday.

**Comment by clason** [comment]
> @theHamsta what is the usual workflow here? For these kinds of PR, do we need to ping the maintainer of the grammar?

Yes, that is the correct etiquette.

**Comment by matthias-Q** [comment]
@DerekStride after your last approval, I have added `@float` query using a regex. Can you check that?

File: queries/sql/highlights.scm [hunk 2]
```diff
@@ -27,13 +27,64 @@
...
+
 (literal) @string
 (comment) @comment
 (marginalia) @comment
 
 ((literal) @number
  (#lua-match? @number "^%d+$"))
 
+((literal) @float
+ (#lua-match? @float "^[-]?%d*.?%d*$"))
```
Comment (DerekStride) [inline queries/sql/highlights.scm:43]:
```suggestion
 (#lua-match? @float "^[-+]?%d*\.%d*$"))
```

The `.` must be present so we need to remove the `?` and I think make it `\.` so it matches a literal `.` character and not any character.

Something like `+1.1` might be valid too so adding the `+` to the start of the regex would be good.

Comment (matthias-Q) [reply queries/sql/highlights.scm:43]: The `+` is not supported by the grammar at the moment. I will leave this out and make the other changes.

**Comment by clason** [comment]
Please rebase and squash your commits.

**Pull request merged**